### ISSUE TITLE
Feature request: Allow to set severity for `ruff check --isolated` e.g. for ci pipeline's code quality as it defaults to `major` making code quality reports hard to decipher

### ISSUE THREAD

**Issue by k0pernikus** [astral-sh/ruff#22801]
I want my ci gitlab pipeline to do two tings with ruff:

1) Error on lint issues, merge requests are now forbidden, must be fixed:

```
uvx ruff check --output-format=gitlab --output-file=$CQ_REPORT .
```

Severity of major is fine.

2) Now I also want to /inform/ about usage of todo, fixme, etc. via:

```
 uvx ruff check --isolated --select FIX,TD006  --output-format=gitlab
```

Severity of major is not ok, as these are information. They serve a purpose and are ok to be in the code base, yet I want to give developers an overview of what they introduced. (Sometimes they might just have actually forgotten to implement somehting, sometimes this will be a long standing comment. That is ok.)

---

This is how a incident is reported currently:

```json
  {
    "check_name": "FIX002",
    "description": "FIX002: Line contains TODO, consider resolving the issue",
    "severity": "major",
    "fingerprint": "3887a1f7ce896661",
    "location": {
      "path": "portal\\gui\\tests\\address_test.py",
      "positions": {
        "begin": {
          "line": 21,
          "column": 7
        },
        "end": {
          "line": 21,
          "column": 11
        }
      }
    }
  }
```

---

This makes it hard in my gitlab ci to distinguish:

<img width="790" height="315" alt="Image" src="https://github.com/user-attachments/assets/3e6a71bf-5c3e-4352-aad4-fdeee88b4b65" />

I want to set a lower severity, yet I don't see any flag / option for it.

For this use case, I think I will sed / jq replace the file's content, yet it'd be great that one can set the `severity` esp. if ruff is run with `--isolated` flag.

---

For what's worth, this is my gitlab as I intend to build the `lint` stage:

```yml
.linter-base:
  stage: lint
  image: astral/uv:alpine
  needs: []

.linter-report:
  extends: .linter-base
  variables:
    CQ_REPORT: gl-code-quality-${CI_JOB_NAME_SLUG}.json
    RUFF_ARGS: ''
  script:
    - set -x
    - uvx ruff check $RUFF_ARGS --output-format=gitlab --output-file=$CQ_REPORT .
  artifacts:
    when: always
    reports:
      codequality: $CQ_REPORT

python_code_style_check:
  extends: .linter-report

python_code_todo_and_fixme_finder:
  extends: .linter-report
  allow_failure: true
  variables:
    RUFF_ARGS: --isolated --select FIX,TD006

python_format_check:
  extends: .linter-base
  allow_failure: true
  script:
    - uvx ruff format --check .
  rules:
    - if: $CI_PIPELINE_SOURCE == "merge_request_event" && $CI_MERGE_REQUEST_TARGET_BRANCH_NAME
        == $CI_DEFAULT_BRANCH
    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
````

**Comment by MichaReiser** [comment]
Thank you. This is https://github.com/astral-sh/ruff/issues/1256

**Comment by k0pernikus** [comment]
I overwrite the severity now manually via:

```
uvx ruff check --isolated --select FIX,TD006 --output-format=gitlab  . > report.json || true
uvx jello '[item.update({"severity": "info"}) or item if item.get("severity") == "major" else item for item in _]' < report.json > $CQ_REPORT
```

and it now renders nicely:

<img width="696" height="361" alt="Image" src="https://github.com/user-attachments/assets/e6950ca2-cbc4-4730-9e3f-52c21f6634f8" />

Just be aware that depending on your rule selection, on gitlab at least, the most current reports wins over the other. (Since my reports work on completely different rule sets, they can't intermix and hence can run in parallel).
